server/static/dashboard/collab.js

```diff
@@ -7,42 +10,486 @@ collab.fetch(function(err) {
+ */
+function showFiles(files, extraArgs) {
   var list = document.querySelector('#files');
   files.sort(function(a, b) { return a.data.filepath.localeCompare(b.data.filepath); });
   files.forEach(function(file) {
     var item = document.importNode(document.querySelector('#file').content, true);
+    
     var heading = item.querySelector('h4');
     heading.textContent = file.data.filepath;
     var diff = item.querySelector('.diff code');
     list.appendChild(item);
-    
+
     $.ajax('/baseline/' + project + '/' + file.data.filepath).done(function(baseline) {
+      // Save data for regex updating
+      // TODO: Better way to do this?
+      $(diff).data('baseline', baseline);
+      $(diff).data('text', file.data.text); // TODO: update this data on update
+      $(diff).data('filepath', file.data.filepath);
+
+      var extraArgsForFile = Object.assign({'filepath': file.data.filepath}, extraArgs);
+
       if (cutoff) {
         $.ajax('/historical/' + project + '/' + collabid + '/' + file.data.filepath + '/' + cutoff).done(function(historical) {
-          updateDiff(diff, baseline, historical.data ? historical.data.text : undefined);
+
+          // Used for regex updating
+          if (historical.data) {
+            $(diff).data('text', historical.data.text); // TODO: update this data on update
+          }
+
+          updateFunction(diff, baseline, historical.data ? historical.data.text : undefined, extraArgsForFile);
+
         }).fail(function(req, status, err) {
           diff.textContent = 'Error fetching code: ' + errorToString(req.responseJSON, status, err);
         });
       } else {
         file.subscribe(function() {
-          updateDiff(diff, baseline, file.data.text);
-          file.on('op', function() {
-            updateDiff(diff, baseline, file.data.text);
+          updateFunction(diff, baseline, file.data.text, extraArgsForFile);
+          file.on('op', function(op) {
+            extraArgs["op"] = op;
+            updateFunction(diff, baseline, file.data.text, extraArgsForFile);
           });
         });
       }
     }).fail(function(req, status, err) {
       diff.textContent = 'Error fetching baseline: ' + errorToString(req.responseJSON, status, err);
     });
+
   });
-});
+}
 
-function updateDiff(node, baseline, text) {
+
+
+/////////////////////////////////////////////
+///////////// UPDATE FUNCTIONS //////////////
+// Displays DOM differently for different visualizations.
+
+function updateFunction(node, baseline, text, extraArgs) {
   if (baseline === undefined || text === undefined) { return; }
   node.innerHTML = '';
+  // TODO: This causes page to seem as if it's refreshing every
+  //   time you change the regex, and causes you to lose whatever
+  //   spot in the files you were at
+
+  var threshold = parameters['threshold'];
+  var regexes = parameters['regexes'];
+  var hideCommon = parameters['hideCommonPrefixes'];
+
+  if (!threshold) {
+    // No treshold means no total diff
```

Comment:
"total" → maybe "chunked"? A consistent descriptive name.

Reply:
Could remove this comment because the if condition is now more clear.